cmd/api-server/main.go

```diff
@@ -114,3 +174,73 @@ func runServer(c *cli.Context) error {
 
 	return nil
 }
+
+// loadConfig loads configuration from CLI context
+func loadConfig(c *cli.Context) Config {
+	var cfg Config
+
+	// API configuration
+	cfg.API.Key = c.String("apiKey")
+	cfg.API.AlgorithmPath = c.String("algorithmPath")
+	cfg.API.Host = c.String("host")
+	cfg.API.Port = c.String("port")
+
+	// Auth configuration
+	cfg.Auth.Domain = c.String("auth0Domain")
+	cfg.Auth.ClientID = c.String("auth0ClientId")
+	cfg.Auth.ClientSecret = c.String("auth0ClientSecret")
+	cfg.Auth.CallbackURL = c.String("auth0CallbackUrl")
+	cfg.Auth.HmacSecret = c.String("hmacSecret")
+	cfg.Auth.ApiAudience = c.String("auth0ApiAudience")
+
+	// DB configuration
+	cfg.DB.DSN = c.String("dsn")
+
+	return cfg
+}
```

Comment:
Usually, this kind of stuff, besides the DSN I would put in a config file instead of a million flags :p But up to you